spec/core/warning/warn_spec.rb

```diff
@@ -0,0 +1,137 @@
+require_relative '../../spec_helper'
+
+describe "Warning.warn" do
+  it "complains" do
+    -> {
+      Warning.warn("Chunky bacon!")
+    }.should complain("Chunky bacon!")
+  end
+
+  it "does not add a newline" do
+    ruby_exe("Warning.warn('test')", args: "2>&1").should == "test"
+  end
+
+  it "returns nil" do
+    ruby_exe("p Warning.warn('test')", args: "2>&1").should == "testnil\n"
+  end
+
+  it "extends itself" do
+    Warning.singleton_class.ancestors.should include(Warning)
+  end
+
+  it "has Warning as the method owner" do
+    ruby_exe("p Warning.method(:warn).owner").should == "Warning\n"
+  end
+
+  # NATFIXME: Implement IO#sync=
+  # it "can be overridden" do
+  #   code = <<-RUBY
+  #     $stdout.sync = true
+  #     $stderr.sync = true
+  #     def Warning.warn(msg)
+  #       if msg.start_with?("A")
+  #         puts msg.upcase
+  #       else
+  #         super
+  #       end
+  #     end
+  #     Warning.warn("A warning!")
+  #     Warning.warn("warning from stderr\n")
+  #   RUBY
+  #   ruby_exe(code, args: "2>&1").should == %Q[A WARNING!\nwarning from stderr\n]
+  # end
+
+  # NATFIXME: Warning for hash key duplicated and overwritten
+  # it "is called by parser warnings" do
+  #   Warning.should_receive(:warn)
+  #   verbose = $VERBOSE
+  #   $VERBOSE = false
+  #   begin
+  #     eval "{ key: :value, key: :value2 }"
+  #   ensure
+  #     $VERBOSE = verbose
+  #   end
+  # end
```

Comment:
It should be possible to use the `NATFIXME` block call instead:
```ruby
it "is called by parser warnings" do
  NATFIXME 'Warning for hash key duplicated and overwritten' do
    Warning.should_receive(:warn)
    # Rest of the test
  end
end
```
This way the tests expect a failure, and we'll get a notification as soon as it is fixed.

Reply:
I tried that originally but it fails with `Issue has been fixed, please remove or update the NATFIXME marker`

Reply:
I've seen that Schrödingers Fixme before, I would guess we're having problems with identifying the `should_receive` in the block.
In this case, it's easier to disable the block by replacing `it` with `xit`, that makes it show up as a skipped test. This time I actually tested it to see if it works ;)

Reply:
I've updated the code to use `xit`